flaml/autogen/agent/math_user_proxy_agent.py

```diff
@@ -0,0 +1,435 @@
+from .user_proxy_agent import UserProxyAgent
+from flaml.autogen.code_utils import UNKNOWN, extract_code, execute_code, infer_lang
+from flaml.autogen.math_utils import get_answer
+from collections import defaultdict
+import re
+import os
+from pydantic import BaseModel, Extra, root_validator
+from typing import Any, Dict, Optional
+from time import sleep
+
+
+PROMPTS = {
```

Comment:
I wonder if it's better to move this inside the class.

Reply:
The class will look chunky with this inside. This is basically an application of the userproxyagent, so I think it is ok to put it outside the class.